### PR TITLE
Add multiset count APIs

### PR DESCRIPTION
**Pull request by PointKernel**
Contributes to #462

This PR adds a `multiset::count` host-bulk API.

### CODE REVIEW

**Comment by PointKernel** [comment]
`cg::reduce` can generate more effective SASS but got an illegal memory access error when trying to read the final global output. To be investigated.

File: include/cuco/detail/open_addressing/kernels.cuh [hunk 1]
```diff
@@ -329,6 +329,60 @@ CUCO_KERNEL void find(InputIt first, cuco::detail::index_type n, OutputIt output
   }
 }
 
+/**
+ * @brief Inserts all elements in the range `[first, first + n)` and returns the number of
+ * successful insertions if `pred` of the corresponding stencil returns true.
+ *
+ * @note If multiple elements in `[first, first + n)` compare equal, it is unspecified which element
+ * is inserted.
+ * @note The key `*(first + i)` is inserted if `pred( *(stencil + i) )` returns true.
```
Comment (sleeepyjack) [inline include/cuco/detail/open_addressing/kernels.cuh]: This docs relate to `insert` rather than `count`

File: include/cuco/operator.hpp [hunk 2]
```diff
@@ -50,6 +50,12 @@ struct erase_tag {
 struct contains_tag {
 } inline constexpr contains;  ///< `cuco::contains` operator
 
+/**
+ * @brief `count` operator tag
+ */
+struct count_tag {
```
Comment (sleeepyjack) [inline include/cuco/operator.hpp:56]: Nit: Should we rename this to `count/find/*_operator` in a future PR?

File: include/cuco/detail/open_addressing/kernels.cuh (hunk 1 above)
Comment (PointKernel) [reply include/cuco/detail/open_addressing/kernels.cuh]: updated

File: include/cuco/operator.hpp (hunk 2 above)
Comment (PointKernel) [reply include/cuco/operator.hpp:56]: Are you suggesting to replace `*_tag` with `*_operator`?

Comment (sleeepyjack) [reply include/cuco/operator.hpp:56]: Yeah, I find my intial choice with "_tag" might not be ideal

Comment (sleeepyjack) [reply include/cuco/operator.hpp:56]: If course that's just a general remark - no action item for this PR

Comment (PointKernel) [reply include/cuco/operator.hpp:56]: It's like the `empty_value` v.s. `empty_payload` discussion we just had. `*_tag` looks fine to me.

**Review by sleeepyjack — APPROVED** [review]

**Pull request merged**